gradle-plugin-testing/src/main/java/com/palantir/gradle/plugintesting/PluginTestingPlugin.java

```diff
@@ -16,13 +16,94 @@
 
 package com.palantir.gradle.plugintesting;
 
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.gradle.api.Action;
+import org.gradle.api.NamedDomainObjectProvider;
 import org.gradle.api.Plugin;
 import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.artifacts.ModuleDependency;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.SourceSetContainer;
+import org.gradle.api.tasks.testing.Test;
 
 public class PluginTestingPlugin implements Plugin<Project> {
+    /**
+     * Used in tests to pick up the current version of this plugin.
+     */
+    static final String PLUGIN_VERSION_PROPERTY_NAME = "pluginTestingPluginVersion";
 
+    /**
+     * Applies the plugin to the given project.
+     */
     @Override
-    public final void apply(Project project) {
-        project.getTasks().register("plugintestingTask", PluginTestingTask.class);
+    public void apply(Project project) {
+        PluginTestingExtension testUtilsExt =
+                project.getExtensions().create(PluginTestingExtension.EXTENSION_NAME, PluginTestingExtension.class);
+
+        addTestDependency(project);
+
+        SourceSetContainer sourceSetContainer = project.getExtensions().getByType(SourceSetContainer.class);
+        SourceSet sourceSet = sourceSetContainer.getByName(SourceSet.TEST_SOURCE_SET_NAME);
+        NamedDomainObjectProvider<Configuration> testRuntimeConfig =
+                project.getConfigurations().named(sourceSet.getRuntimeClasspathConfigurationName());
+
+        project.getTasks().withType(Test.class).configureEach(test -> {
+            // need to use the doFirst so that any custom settings on the extension are applied before reading
+            // the values and setting the system properties.
+            Action<Task> action = new Action<>() {
+                @Override
+                public void execute(Task _task) {
+                    // add system property for all test dependencies so that TestDepVersions can resolve them
+                    Set<String> depSet = getDependencyStrings(testRuntimeConfig.get());
```

Comment:
ya - I talked with Callum about the issue of multiple test configurations after the team meeting on Wed.  Our thinking was this will cover the vast majority of gradle plugins because they either have just the standard Test configuration or the testRuntimeClasspath is a base configuration for any other test configuration.  But I agree that if there is a more complete way we can adopt it.  I'm not sure we want to enforce a whole other plugin though.